enhancements/proposals/node-autoreplacement.md

```diff
@@ -0,0 +1,133 @@
+A node scheduled on this instance loses its storage, even though a PVC bound to it is not removed. The node gets replaced automatically.
+
+### Notes/Constraints/Caveats [Optional]
+
+- It would be preferred to use [Immutable ConfigMaps](https://kubernetes.io/docs/concepts/configuration/configmap/#configmap-immutable).
+- Having introduced this feature, we might consider removing the existing functionality of [replacing dead nodes manually](https://operator.docs.scylladb.com/master/nodeoperations/replace_node.html).
+
+### Risks and Mitigations
+
+- The proposed design is based on a fact that `replace-address-first-boot` flag is ignored after bootstrap. Any changes to this behaviour or side effects introduced could result in an undefined behaviour of the enhancement.
+
+## Design Details
+
+To check whether the node is not the first to bootstrap in a cluster and already exists in gossip, the controller uses [`ScyllaClient's Status`](https://github.com/scylladb/scylla-operator/blob/689c0a0a82aeea0fef72b2cd11af0333d1d2e180/pkg/scyllaclient/client.go#L98).
+Client is created with default config. Controller authenticates with the already known token. It communicates with all nodes of a given cluster. The conditions for creating the `ConfigMap` are met if the node's IP is among the returned hosts.
+
+`Sidecar` accesses the `ConfigMap` through a dedicated `Volume`.
+
+### Test Plan
+
+- e2e/integration tests
+     1.
+          - Create a `ScyllaCluster` with 2 nodes
+          - Wait for the `ScyllaCluster` to deploy
+          - Remove first node's PVC
+          - Remove first node's Pod
+          - Await PVC's and Pod's deletion
+          - Wait for the `ScyllaCluster` to redeploy
+          - Verify `ScyllaCluster`
+     2.
+          - Create a `ScyllaCluster` with 2 nodes
+          - Wait for `ScyllaCluster` to deploy
+          - Turn on maintenance mode, Exec `systemctl stop scylla && rm -rf /var/lib/scylla` in first node's Pod
+          - Remove first node's Pod
+          - Await Pod's deletion
+          - Wait for the `ScyllaCluster` to redeploy
+          - Verify `ScyllaCluster`
+    
+More fitting e2e tests could be created once the tests are run against a real distribution instead of Minikube.
+
+### Upgrade / Downgrade Strategy
+
+This enhancement adds a `ConfigMap` per `ScyllaCluster`'s Pod and extends `StatefulSetSpec` with a dedicated `Volume` to access the config.
+It does not cause any drawbacks on a downgrade.
+
+### Version Skew Strategy
+
+This enhancement only requires changes in the Scylla Operator image. There is no version skew involved.
+
+## Implementation History
+
+- 1.5, enhancement proposal created
+
+## Drawbacks
+
+- Controller needs to make API calls to Scylla nodes.
```

Comment:
if there is a mitigation, like it's very infrequent, it's worth mentioning it as well